sdk/java-sdk/src/main/java/kalix/javasdk/view/View.java

```diff
@@ -24,6 +24,8 @@
 public abstract class View<S> {
 
   private Optional<UpdateContext> updateContext = Optional.empty();
+  private Optional<S> currentState = Optional.empty();
```

Comment:
I think this makes much less sense than for entities since it is not a state per view like the state for an entity, it's the state for a given subject/entity id update being processed right now. 

So passing it as a parameter would be a more reasonable API for view update methods IMO

Reply:
I don't have a strong opinion on this.. happy to change it to a parameter for the views.
In that case, I guess we also want to enforce the signature of the annotated methods to be `something(TransformedUser, User)`. Would you still call it `currentState` btw? Or is there a better name then? 

 @octonato any thoughts here on using a param instead of `currentState()`?

Reply:
In the protocol-first samples we just call it `state`

Reply:
May work for Views indeed. As long as we don't allow it to be updated from REST calls. We can't have two `@RequestBody`. 

I don't think we should restrict it to always receive the state. We can accept two signatures: 
* `onChange(User): TransformedUser`
* `onChange(TransformedUser, User): TransformedUser`

For Views based on ValueEntities, it will be convenient to simply have `onChange(User): TransformedUser`

The only drawback of using an extra method arg, is that it diverge from the other APIs in the Spring SDK. But worth evaluating it, IMO.

Reply:
For event sourcing, two signatures may be interesting as well, for example:

* `onUserCreated(UserCreated): UserView`
* `onUserEmailUpdated(UserView, EmailUpdated): UserView`